Makefile

```diff
@@ -24,6 +24,31 @@ AUTHORS: .mailmap .git/HEAD
 PKGS_AND_MOCKS := $(shell go list ./... | grep -v /vendor)
 PKGS := $(shell echo $(PKGS_AND_MOCKS) | tr ' ' '\n' | grep -v /mock$)
 
+# Current working environment.  Set these explicitly if you want to cross-compile
+# in the build container (see the build-in-container target):
+
+GOOS?=$(shell go env GOOS)
+GOARCH?=$(shell go env GOARCH)
+VENDOR_HASH=$(shell git hash-object vendor/vendor.json)
+
+# First we create a container that is versioned (via the vendor.json hash) that has
+# all the downloaded vendor dependencies.  Then we use that container as the data
+# volume container across multiple builds, as long as the vendoring dependencies
+# don't chnage.  When vendor.json is updated, a new hash results in a new container
+# which is reused across builds again.  If you bulid this target, vendor/ directory
+# will not be changed.  Instead the vendored files will be in the container's fs namespace.
+build-in-container:
+	@echo "+ $@"
+	docker build -t infrakit-build:${VENDOR_HASH} -f ${CURDIR}/dockerfiles/Dockerfile.build .
+	-docker run --name infrakit-build-${VENDOR_HASH} \
+		-e GOOS=${GOOS} -e GOARCCH=${GOARCH} \
+		infrakit-build:${VENDOR_HASH}
```

Comment:
An alternative implementation that's a little quicker and doesn't require volumes:
```diff
diff --git a/.dockerignore b/.dockerignore
new file mode 100644
index 0000000..d097005
--- /dev/null
+++ b/.dockerignore
@@ -0,0 +1,4 @@
+build
+dockerfiles
+vendor
+!vendor/vendor.json
diff --git a/Makefile b/Makefile
index da5ebd0..72dc343 100644
--- a/Makefile
+++ b/Makefile
@@ -49,6 +49,14 @@ build-in-container:
                --volumes-from infrakit-build-${VENDOR_HASH} \
                infrakit-build:${VENDOR_HASH} make build-binaries
 
+build-in-container-simple:
+       @echo "+ $@"
+       docker build -t infrakit-build -f ${CURDIR}/dockerfiles/Dockerfile.build-simple .
+       docker run --rm \
+               -e GOOS=${GOOS} -e GOARCCH=${GOARCH} \
+               -v ${CURDIR}/build:/go/src/github.com/docker/infrakit/build \
+               infrakit-build make build-binaries
+
 vet:
        @echo "+ $@"
        @go vet $(PKGS)
diff --git a/dockerfiles/Dockerfile.build-simple b/dockerfiles/Dockerfile.build-simple
new file mode 100644
index 0000000..a725800
--- /dev/null
+++ b/dockerfiles/Dockerfile.build-simple
@@ -0,0 +1,9 @@
+FROM infrakit/devtool:go-1.7.1
+
+WORKDIR /go/src/github.com/docker/infrakit
+
+COPY Makefile /go/src/github.com/docker/infrakit/Makefile
+COPY vendor/vendor.json /go/src/github.com/docker/infrakit/vendor/vendor.json
+RUN make vendor-sync
+
+COPY . /go/src/github.com/docker/infrakit
```